File: backend/src/models/Evidence.ts
import mongoose, { Schema, type Document } from 'mongoose';

export interface IEvidence extends Document {
  incidentId: mongoose.Types.ObjectId;
  uploadedBy: mongoose.Types.ObjectId;
  fileName: string;
  filePath: string;
  mimeType: string;
  fileSize: number;
  validationStatus: 'PENDING' | 'VALID' | 'REJECTED';
  metadata?: Record<string, unknown>;
  uploadedAt: Date;
}

const evidenceSchema = new Schema<IEvidence>(
  {
    incidentId: { type: Schema.Types.ObjectId, ref: 'Incident', required: true },
    uploadedBy: { type: Schema.Types.ObjectId, ref: 'User', required: true },
    fileName: { type: String, required: true },
    filePath: { type: String, required: true },
    mimeType: { type: String, required: true },
    fileSize: { type: Number, required: true },
    validationStatus: {
      type: String,
      enum: ['PENDING', 'VALID', 'REJECTED'],
      default: 'PENDING'
    },
    metadata: { type: Schema.Types.Mixed, default: {} },
    uploadedAt: { type: Date, default: Date.now }
  },
  { timestamps: true }
);

evidenceSchema.index({ incidentId: 1, uploadedAt: -1 });

evidenceSchema.index({ uploadedBy: 1 });

export const Evidence = mongoose.model<IEvidence>('Evidence', evidenceSchema);

File: backend/src/main/java/com/raava/model/Evidence.java
package com.raava.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.Instant;

@Document(collection = "evidence")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Evidence {
    @Id
    private String id;

    @DBRef(lazy = true)
    private Incident incident;

    @DBRef(lazy = true)
    private User uploadedBy;

    private String fileName;
    private String filePath;
    private String mimeType;
    private long fileSize;

    private ValidationStatus validationStatus = ValidationStatus.PENDING;
    private String metadata;

    @Indexed
    private Instant createdAt;

    @Indexed
    private Instant updatedAt;

    public enum ValidationStatus { PENDING, VALIDATED, REJECTED }
}
